etest_test/profile_test.py

```diff
@@ -0,0 +1,153 @@
+"""Tests for etest.profile."""
+
+import logging
+from typing import Any, Dict
+
+import pytest
+from hypothesis import assume, given
+from hypothesis import strategies as st
+from pytest_mock import MockerFixture
+
+import etest.profile as sut
+
+logger = logging.getLogger(__name__)
+
+
+_valid_architectures = ["amd64", "x86", "armv5", "armv6", "armv7", "arm64", "ppc64"]
+_valid_systemd_architectures = ["amd64", "x86", "arm64", "ppc64"]
+_libc_value = ["glibc", "musl", "uclibc"]
+
+
+@st.composite
+def non_base_profiles(draw: Any) -> Dict[str, Any]:
+    """Strategy for generating non-base profiles."""
+    # Draw values
+    profile = {}
+    profile["libc"] = draw(st.sampled_from(_libc_value))
+    profile["hardened"] = draw(st.booleans())
+    profile["multilib"] = draw(st.booleans())
+    profile["systemd"] = draw(st.booleans())
+
+    # Ensure it isn't a base profile
+    assume(profile != {"libc": "glibc", "hardened": False, "multilib": True, "systemd": False})
+
+    return profile
+
+
+@pytest.mark.parametrize("arch", _valid_architectures)
+def test_base_arch(arch: str) -> None:
+    """Ensure the base profiles work."""
+    result = sut.Profile(False, arch, "glibc", False, True, False)
+
+    # Check profile strings
+    assert result.profile == arch
```

Comment:
This is true by definition unless I'm missing something.

Reply:
This tests it hasnt added something extra by mistake, like a dash or an option. Let me know if that makes sense.

Reply:
How would something get added to this?  Is the definition of profile not just arch?  That's what I'm reading this test as.  It's just testing assignment doesn't add anything so it's not worth testing since assignment does exactly what you expect.